src/js/angular/import/controllers/import-view.controller.js

```diff
@@ -390,7 +393,7 @@ importViewModule.controller('ImportViewCtrl', ['$scope', 'toastr', '$interval',
                 $scope.savedSettings = _.mapKeys(_.filter($scope.files, 'parserSettings'), 'name');
             }).error(function (data) {
                 toastr.warning($translate.instant('import.error.could.not.get.files', {data: getError(data)}));
-            });
+            }).finally(() => ImportContextService.updateShowLoader(false));
```

Comment:
This is a bit odd. I don't see you setting the loader to true in this functinon, but you set it to false. I think it is better to have only one method responsible fo the loader states change. You can return the promise and put a finally at the other end.

Reply:
Or set it true here before the promise. Whichever makes more sense.

Reply:
This method is called at regular intervals. We want the loader to be displayed when the view is loaded or the tab is changed. When this happens, the showLoader variable is set to true.

Reply:
I agree with @yordanalexandrov that this is not the best solution and the first thing he proposed would be a better approach, a.k.a. return the promise form updateListHttp method and hide the loader in the caller

Reply:
done